aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -143,11 +143,17 @@ void scatter_meta_impl(
 
 TORCH_META_FUNC2(scatter, src)
 (const Tensor& self, int64_t dim, const Tensor& index, const Tensor& src) {
+  // See note [Writing Nondeterministic Operations]
+  // Nondeterministic when index contains duplicate entries
+  at::globalContext().alertNotDeterministic("scatter_src_out");
```

Comment:
In the future should we think about letting the users know information like, "This operation is deterministic if the input indices are unique, and this can be validated by..."? 

Or is it preferable to just implement the best validation of unique input indices as the deterministic version of scatter?

Reply:
Yeah it might be a good idea to let users know the conditions that make an operation nondeterministic.

> implement the best validation of unique input indices as the deterministic version of scatter

Do you mean something like this?

```
if (using deterministic algorithms) and (index contains duplicate entries):
    throw error
else:
    run operation
```